src/main/java/com/proofpoint/discovery/store/HttpRemoteStore.java

```diff
@@ -197,6 +207,20 @@ private static String getHostPort(ServiceDescriptor descriptor)
     @Override
     public void put(Entry entry)
     {
+        if (generalPoolLegacyReplicationMode == ReplicationMode.PHASE_ONE) {
+            List<Service> services = entry.getValue();
+            if (services != null) {
+                services = services.stream()
+                        .map(service -> {
```

Comment:
You can write it like that so that stream API helps to eliminate lots of curly braces:

```
  .filter(service.getPool().equals(generalPoolMapTarget))
  .map(service -> Service.copyOf(service).setPool("general").build())
  .collect(toImmutableList());
```

Reply:
That would incorrectly remove services not in the generalPoolMapTarget from the resulting list